src/intervalmap.rs

```diff
@@ -719,423 +663,149 @@ where
         R: 'a,
         F: FnOnce(&'a mut Node<T, V, Ix>) -> R,
     {
-        let idx = self.nodes[node.index()].right().index();
+        let idx = self.nodes[node_idx.index()].right().index();
         op(&mut self.nodes[idx])
     }
 
-    fn parent_mut<F, R>(&'a mut self, node: NodeIndex<Ix>, op: F) -> R
+    fn parent_mut<F, R>(&'a mut self, node_idx: NodeIndex<Ix>, op: F) -> R
     where
         R: 'a,
         F: FnOnce(&'a mut Node<T, V, Ix>) -> R,
     {
-        let idx = self.nodes[node.index()].parent().index();
+        let idx = self.nodes[node_idx.index()].parent().index();
         op(&mut self.nodes[idx])
     }
 
-    fn grand_parent_mut<F, R>(&'a mut self, node: NodeIndex<Ix>, op: F) -> R
+    fn grand_parent_mut<F, R>(&'a mut self, node_idx: NodeIndex<Ix>, op: F) -> R
     where
         R: 'a,
         F: FnOnce(&'a mut Node<T, V, Ix>) -> R,
     {
-        let parent_idx = self.nodes[node.index()].parent().index();
+        let parent_idx = self.nodes[node_idx.index()].parent().index();
         let grand_parent_idx = self.nodes[parent_idx].parent().index();
         op(&mut self.nodes[grand_parent_idx])
     }
 
-    fn max(&self, node: NodeIndex<Ix>) -> Option<&T> {
-        let max_index = self.nodes[node.index()].max_index?.index();
+    pub(crate) fn max(&self, node_idx: NodeIndex<Ix>) -> Option<&T> {
+        let max_index = self.nodes[node_idx.index()].max_index?.index();
         self.nodes[max_index].interval.as_ref().map(|i| &i.high)
     }
 }
-
-/// An iterator over the entries of a `IntervalMap`.
-#[derive(Debug)]
-pub struct Iter<'a, T, V, Ix> {
-    /// Reference to the map
-    map_ref: &'a IntervalMap<T, V, Ix>,
-    /// Stack for iteration
-    stack: Option<Vec<NodeIndex<Ix>>>,
-}
-
-impl<T, V, Ix> Iter<'_, T, V, Ix>
-where
-    Ix: IndexType,
-{
-    /// Initializes the stack
-    fn init_stack(&mut self) {
-        self.stack = Some(Self::left_link(self.map_ref, self.map_ref.root));
-    }
-
-    /// Pushes a link of nodes on the left to stack.
-    fn left_link(map_ref: &IntervalMap<T, V, Ix>, mut x: NodeIndex<Ix>) -> Vec<NodeIndex<Ix>> {
-        let mut nodes = vec![];
-        while !map_ref.node_ref(x, Node::is_sentinel) {
-            nodes.push(x);
-            x = map_ref.node_ref(x, Node::left);
-        }
-        nodes
-    }
-}
-
-impl<'a, T, V, Ix> Iterator for Iter<'a, T, V, Ix>
-where
-    Ix: IndexType,
-{
-    type Item = (&'a Interval<T>, &'a V);
-
-    #[inline]
-    fn next(&mut self) -> Option<Self::Item> {
-        if self.stack.is_none() {
-            self.init_stack();
-        }
-        let stack = self.stack.as_mut().unwrap();
-        if stack.is_empty() {
-            return None;
-        }
-        let x = stack.pop().unwrap();
-        stack.extend(Self::left_link(
-            self.map_ref,
-            self.map_ref.node_ref(x, Node::right),
-        ));
-        Some(self.map_ref.node_ref(x, |xn| (xn.interval(), xn.value())))
-    }
-}
-
-#[cfg(test)]
-mod test {
```

Comment:
Why removes the tests?

Reply:
These tests will be presented at next pr.